diff --git a/btif/src/btif_media_task.c b/btif/src/btif_media_task.c
index de34818..d75cda8 100644
--- a/btif/src/btif_media_task.c
+++ b/btif/src/btif_media_task.c
@@ -2458,13 +2458,14 @@ static UINT8 btif_get_num_aa_frame(void)
 
             /* calculate nbr of frames pending for this media tick */
             result = btif_media_cb.media_feeding_state.pcm.counter/pcm_bytes_per_frame;
+
+            btif_media_cb.media_feeding_state.pcm.counter -= result*pcm_bytes_per_frame;
             if (result > MAX_PCM_FRAME_NUM_PER_TICK)
             {
                 APPL_TRACE_WARNING("%s() - Limiting frames to be sent from %d to %d"
                     , __FUNCTION__, result, MAX_PCM_FRAME_NUM_PER_TICK);
                 result = MAX_PCM_FRAME_NUM_PER_TICK;
             }
-            btif_media_cb.media_feeding_state.pcm.counter -= result*pcm_bytes_per_frame;
 
             VERBOSE("WRITE %d FRAMES", result);
         }
